content/en/docs/topics/connecting-to-an-sqlite-database.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Connecting to an SQLite database
+description: Connect your Spin App to an external SQLite database
+date: 2024-07-17
+categories: [Spin Operator]
+tags: [Tutorials]
+weight: 13
+---
+
+Spin applications can utilize a [standardized API for persisting data in an SQLite database](https://developer.fermyon.com/spin/v2/sqlite-api-guide). A default database is created by the Spin runtime on the local filesystem, which is great for getting an application up and running. However, this on-disk solution may not be preferable for an app running in the context of SpinKube, where apps are often scaled beyond just one replica.
+
+Thankfully, Spin supports configuring an application with an [external SQLite database provider via runtime configuration](https://developer.fermyon.com/spin/v2/dynamic-configuration#libsql-storage-provider). External providers include any [libSQL](https://libsql.org/) databases that can be accessed over HTTPS.
+
+## Prerequisites
+
+To follow along with this tutorial, you'll need:
+
+- A Kubernetes cluster running SpinKube. See the [Installation](../install/_index.md) guides for more information.
+- The [kubectl CLI](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- The [spin CLI](https://developer.fermyon.com/spin/v2/install )
+
+## Build and publish the Spin application
+
+For this tutorial, we'll use the [HTTP CRUD Go SQLite](https://github.com/fermyon/enterprise-architectures-and-patterns/tree/main/http-crud-go-sqlite) sample application. It is a Go-based app implementing CRUD (Create, Read, Update, Delete) operations via the SQLite API.
+
+First, clone the repository locally and navigate to the `http-crud-go-sqlite` directory:
+
+```bash
+git clone git@github.com:fermyon/enterprise-architectures-and-patterns.git
+cd enterprise-architectures-and-patterns/http-crud-go-sqlite
+```
+
+Now, build and push the application to a registry you have access to. Here we'll use [ttl.sh](https://ttl.sh):
+
+```bash
+spin build
+spin registry push ttl.sh/http-crud-go-sqlite:1h
```

Comment:
In the event that multiple people run the tutorial at the same time, would it be better to create a unique image name? e.g. as shown in ttl.sh's landing page:

```
export IMAGE_NAME=ttl.sh/$(uuidgen):1h
spin build
spin registry push ${IMAGE_NAME}
```